### DISCUSSION TITLE
Pre-provision Shell selection and Storage Connection - Feature Request

### DISCUSSION THREAD

**Discussion by movinalot** [Azure/CloudShell#212] [Ideas]
## To Reproduce

Initial Cloud Shell provisioning

```
NA
```

## Observed Behavior

User is prompted to select shell and storage

```
NA
```

## Expected behavior

Works as expected.

## Is this specific to Cloud Shell?

This is specific to Cloud Shell.

## Interface information

Accessing Cloud Shell - https://shell.azure.com or https://portal.azure.com

## Additional context

Feature Request.

Using Terraform I create training environments where users use Cloud Shell to run the training exercises.

The Terraform is used to create storage that is intended to be used for Cloud Shell. 

I would like to be able to automate the shell selection process and storage connection process, so that when a user connects to Cloud Shell they are taken directly to the shell prompt.

APIs that can be called from Terraform, other IaC tooling, shell script, etc.  to support this would be great.

**Comment by maertendMSFT** [comment]
This request is currently in our backlog and is not currently scheduled.  Please upvote the initial post if this is valuable to others.
